conn.go

```diff
@@ -38,14 +40,28 @@ func (c *conn) IsClosed() bool {
 
 // OpenStream creates a new stream.
 func (c *conn) OpenStream(ctx context.Context) (mux.MuxedStream, error) {
+	scope, err := c.rcmgr.OpenStream(c.remotePeerID, network.DirOutbound)
+	if err != nil {
+		return nil, err
+	}
 	qstr, err := c.sess.OpenStreamSync(ctx)
+	if err != nil {
+		scope.Done()
+		return nil, err
+	}
 	return &stream{Stream: qstr}, err
 }
 
 // AcceptStream accepts a stream opened by the other side.
 func (c *conn) AcceptStream() (mux.MuxedStream, error) {
 	qstr, err := c.sess.AcceptStream(context.Background())
-	return &stream{Stream: qstr}, err
+	if err != nil {
+		return nil, err
+	}
+	if _, err := c.rcmgr.OpenStream(c.remotePeerID, network.DirInbound); err != nil {
+		return nil, err
+	}
+	return &stream{Stream: qstr}, nil
```

Comment:
shouldnt it keep the stream scope?

Reply:
we leak it otherwise

Reply:
The `MuxedStream` doesn't have any scope.
This is probably the wrong layer to do it, this should be done in the swarm.

Reply:
This is now done in the swarm, see https://github.com/libp2p/go-libp2p-swarm/pull/308.